lib/utils/stream.js

```diff
@@ -12,16 +12,19 @@
 var Stream = function() {
   this.init = function() {
     var listeners = {};
+    var sentinels = {};
     /**
      * Add a listener for a specified event type.
      * @param type {string} the event name
      * @param listener {function} the callback to be invoked when an event of
      * the specified type occurs
```

Comment:
maybe add info on the sentinel variable in the doc comments (you can keep the same reversed style if you want, since it will keep it consistent even though its technically wrong)